uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestAttachment.php

```diff
@@ -72,4 +73,28 @@ private function callAttachmentAPI($action, $params) {
 
     return civicrm_api3('Attachment', $action, $params);
   }
+
+  /**
+   * Uses the Attachment API to retrieve attachments associated with a LeaveRequest.
+   * It ensures that the current user can only retrieve Leave attachments for the
+   * Leave requests linked to the contacts the user has access to. The admin can
+   * retrieve all Leave attachments for all contacts.
+   *
+   * @param array $params
+   *
+   * @return array
+   */
+  public function get($params) {
+    $leaveRequestID = isset($params['entity_id']) ? $params['entity_id'] : '';
+    $leaveRequest = LeaveRequest::findById($leaveRequestID);
+    $leaveManagerService = new LeaveManagerService();
+    $leaveRequestRights = new LeaveRequestRights($leaveManagerService);
+    $accessibleContacts = $leaveRequestRights->getLeaveContactsCurrentUserHasAccessTo();
+
+    if ($leaveManagerService->currentUserIsAdmin() || in_array($leaveRequest->contact_id, $accessibleContacts)) {
+      return $this->callAttachmentAPI('get', $params);
+    }
+
+    return [];
+  }
```

Comment:
Please also write tests for this method

Reply:
I was thinking it will be a duplication of the tests for the LeaveRequest.getAttachments API, since this API uses this method directly.

I could modify this method to accept params and also the LeaveRequestRights service so that I can mock that and pass it in the function but I would still have to create leave requests and all. What do you suggest?

Reply:
Unfortunately, because of how the API works, there's no way to avoid this duplication. The tests wouldn't be completely duplicated though, since, as you said, you can mock a few things. In this case it's better to have some duplication than not having tests for this method